src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -330,129 +371,131 @@ int MPL_gavl_tree_free(MPL_gavl_tree_t gavl_tree)
     return mpl_err;
 }
 
+static void gavl_tree_delete_search(uintptr_t addr, uintptr_t len,
+                                    gavl_tree_node_s * cur_node,
+                                    gavl_tree_node_s ** cur_node_ret,
+                                    gavl_tree_stack_s * node_stack)
+{
```

Comment:
Can you summarize the difference of this search function and the previous one (insert_search)? Is it possible to unify them. If necessary, we can add a flag (constant and compiler will optimize it away).

Reply:
`gavl_tree_delete_search` does not have direction parameter to tell which side the new node should be inserted

Reply:
The direction can be easily regained by calling the `cmp_func`, right? So if we omit the direction then both search functions can simply return a node (or NULL), and they can be unified by simply calling it `gavl_search`, right? I understand the extra call to `cmp_func` is a cost, but we are just talking about a single boolean expression here, so I think the benefit of cleaner code is larger.

Also, `cmp_func` as function pointer will incur function overhead. An alternative is to make it a enum flag and turn the cmp_func call to a macro call. Since we are inlining the function and the enum flag will be optimized away.